libvast/vast/format/syslog.hpp

```diff
@@ -0,0 +1,296 @@
+  using attribute = parameters;
+
+  template <class Iterator, class Attribute>
+  bool parse(Iterator& f, const Iterator& l, Attribute& x) const {
+    auto param = parameter_parser{}->*[&](parameter in) {
+      if constexpr (!std::is_same_v<Attribute, unused_type>) {
+        auto& [key, value] = in;
+        x[key] = value;
+      }
+    };
+    auto p = +param;
+    return p(f, l, unused);
+  }
+};
+
+// id and paramters of one structured data element
+using structured_data_element = std::tuple<std::string, parameters>;
+
+// The structured_data_element_parser parses one structured data element
+struct structured_data__element_parser
+  : parser<structured_data__element_parser> {
+  using attribute = structured_data_element;
+
+  template <class Iterator, class Attribute>
+  bool parse(Iterator& f, const Iterator& l, Attribute& x) const {
+    using parsers::print, parsers::rep;
+
+    auto is_sd_name_char = [](char in) {
+      return in != '=' && in != ' ' && in != ']' && in != '"';
+    };
+    auto sd_name = print - ' ';
+    auto sd_name_char = sd_name.with(is_sd_name_char);
+    auto sd_id = rep(sd_name_char, 1, 32);
+    auto params = parameters_parser{};
+    auto p = '[' >> sd_id >> params >> ']';
+    if constexpr (std::is_same_v<Attribute, unused_type>)
+      return p(f, l, unused);
+    else
+      return p(f, l, x);
+  }
+};
+
+// all structured data elements
+using structured_data = vast::map;
+
+// The structured_data_parser parses the structured_data elements of one Syslog
+// message
+struct structured_data_parser : parser<structured_data_parser> {
+  using attribute = structured_data;
+
+  template <class Iterator, class Attribute>
+  bool parse(Iterator& f, const Iterator& l, Attribute& x) const {
+    using namespace parsers;
+
+    auto sd
+      = structured_data__element_parser{}->*[&](structured_data_element in) {
+          if constexpr (!std::is_same_v<Attribute, unused_type>) {
+            auto& [key, value] = in;
+            x[key] = value;
+          }
+        };
+    auto p = maybe_nil(+sd);
+    return p(f, l, unused);
+  }
+};
+
+// the message content of the syslog message
+using message_content = std::string;
+
+// The message_content_parser parses the message component of a Syslog message
+struct message_content_parser : parser<message_content_parser> {
+  using attribute = message_content;
+  template <class Iterator, class Attribute>
+  bool parse(Iterator& f, const Iterator& l, Attribute& x) const {
+    using namespace parsers;
+    using namespace parser_literals;
+    auto bom = "\xEF\xBB\xBF"_p;
+    auto p = (bom >> +any) | (+any) | eoi;
+
+    return p(f, l, x);
+  }
+};
+
+// header, data and msg describe one syslog message
+struct message {
+  header header;
+  structured_data data;
+  caf::optional<message_content> msg;
+};
+
+// the message parses uses the header_parser, structured_datas_parser and the
+// message_content_parser to parse one Syslog message
```

Comment:
```suggestion
/// Parser for Syslog messages.
/// @relates message
```

Reply:
You can apply this style of documentation comment to all other classes in this file as well.